synapse/config/tls.py

```diff
@@ -15,66 +15,99 @@
 
 import os
 import subprocess
+from datetime import datetime
 from hashlib import sha256
 
 from unpaddedbase64 import encode_base64
 
 from OpenSSL import crypto
 
-from ._base import Config
-
-GENERATE_DH_PARAMS = False
+from synapse.config._base import Config
 
 
 class TlsConfig(Config):
     def read_config(self, config):
-        self.tls_certificate = self.read_tls_certificate(
-            config.get("tls_certificate_path")
+
+        acme_config = config.get("acme", {})
+        self.acme_enabled = acme_config.get("enabled", False)
+        self.acme_url = acme_config.get(
+            "url", "https://acme-staging.api.letsencrypt.org/directory"
         )
-        self.tls_certificate_file = config.get("tls_certificate_path")
+        self.acme_client_key = config.get("client_key", self.config_dir_path)
```

Comment:
please could this be called `[acme_]client_key_*path*`, since it's a filename rather than the key itself?